### PR TITLE
update targets

### PR DESCRIPTION
**Pull request by chris-rock**
- fix url helper
- update chef compliance and supermarket plugin
- move core to plugins/target

### CODE REVIEW

File: lib/bundles/inspec-compliance/target.rb [hunk 1]
```diff
@@ -39,11 +39,11 @@ def resolve(target, opts = {})
...
-    def get_profile_name(uri)
+    def self.get_profile_name(uri)
       uri.host + uri.path
     end
 
-    def build_target_url(target)
+    def self.build_target_url(target)
       owner, profile = target.split('/')
       config = Compliance::Configuration.new
       url = "#{config['server']}/owners/%owner_name%/compliance/%profile_name%/tar"
```
Comment (srenatus) [inline lib/bundles/inspec-compliance/target.rb:49]: This is a little weird. Why not just this (instead of line 49-52)

`"#{config['server']}/owners/#{owner}/compliance/#{profile}/tar"`?

I know this isn't part of the change set but fixing it here is as good a time as any ;)

**Comment by arlimus** [comment]
I like the separation into a more plugin-aware structure, that's fantastic! The current system (I introduced) is really confusing.

imho DirHelper is not the right base class for UrlHelper. Looking at the code: It hands everything off to zip or tar anyway, i.e. there is no re-use of DirHelper functionality, except for tricky indirection through zip/tar-helpers; the disadvantage is that it forces you to write every other piece that builds on UrlHelper (like Supermarket and Compliance) in a way that requires class-methods (i.e. another bucket of code-changes).
Worse, the indirection of Zip/Tar-helpers is really hard to follow (took me a while to understand what was wrong in the first place).

I'll update this in a bit...

File: lib/bundles/inspec-compliance/target.rb (hunk 1 above)
Comment (arlimus) [inline lib/bundles/inspec-compliance/target.rb:49]: +1 great idea for clarity

Comment (chris-rock) [inline lib/bundles/inspec-compliance/target.rb:49]: like it

**Comment by chris-rock** [comment]
close in favor for #462
